app/src/ui/rebase/rebase-branch-dialog.tsx

```diff
@@ -0,0 +1,189 @@
+
+  /**
+   * A function that's called when the dialog is dismissed by the user in the
+   * ways described in the Dialog component's dismissable prop.
+   */
+  readonly onDismissed: () => void
+}
+
+interface IRebaseBranchDialogState {
+  /** The currently selected branch. */
+  readonly selectedBranch: Branch | null
+
+  /** The filter text to use in the branch selector */
+  readonly filterText: string
+
+  readonly isRebasing: boolean
+}
+
+/** A component for initating a rebase of the current branch. */
+export class RebaseBranchDialog extends React.Component<
+  IRebaseBranchDialogProps,
+  IRebaseBranchDialogState
+> {
+  public constructor(props: IRebaseBranchDialogProps) {
+    super(props)
+
+    const selectedBranch = this.resolveSelectedBranch()
+
+    this.state = {
+      selectedBranch,
+      filterText: '',
+      isRebasing: false,
+    }
+  }
+
+  private onFilterTextChanged = (filterText: string) => {
+    this.setState({ filterText })
+  }
+
+  private onSelectionChanged = (selectedBranch: Branch | null) => {
+    this.setState({ selectedBranch })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  public render() {
+    const selectedBranch = this.state.selectedBranch
+    const currentBranch = this.props.currentBranch
+
+    const selectedBranchIsNotCurrentBranch =
+      selectedBranch === null ||
+      currentBranch === null ||
+      currentBranch.name === selectedBranch.name
+
+    const disabled = selectedBranchIsNotCurrentBranch
+
+    // the amount of characters to allow before we truncate was chosen arbitrarily
+    const currentBranchName = truncateWithEllipsis(
+      this.props.currentBranch.name,
+      40
+    )
+
+    const loading = this.state.isRebasing ? <Loading /> : null
+
+    return (
+      <Dialog
+        id="rebase"
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.startRebase}
+      >
+        <DialogHeader
+          title={
+            <div className="rebase-dialog-header">
+              Rebase <strong>{currentBranchName}</strong> onto…
+            </div>
+          }
+          dismissable={true}
+          onDismissed={this.props.onDismissed}
+        />
+        <DialogContent>
+          <BranchList
+            allBranches={this.props.allBranches}
+            currentBranch={currentBranch}
+            defaultBranch={this.props.defaultBranch}
+            recentBranches={this.props.recentBranches}
+            filterText={this.state.filterText}
+            onFilterTextChanged={this.onFilterTextChanged}
+            selectedBranch={selectedBranch}
+            onSelectionChanged={this.onSelectionChanged}
+            canCreateNewBranch={false}
+            renderBranch={this.renderBranch}
+          />
+        </DialogContent>
+        <DialogFooter>
+          <ButtonGroup>
+            <Button type="submit" disabled={disabled}>
```

Comment:
Can we please use the `<Dialog disabled=` attribute here instead of explicitly disabling the button?

Reply:
Addressed in 9809329c7d4cef3c5c7ec76dea26592283db2223